Examples/TicTacToe/tic-tac-toe/Sources/AppSwiftUI/AppView.swift

```diff
@@ -12,18 +12,22 @@ public struct AppView: View {
   }
 
   public var body: some View {
-    SwitchStore(self.store) {
-      CaseLet(state: /TicTacToe.State.login, action: TicTacToe.Action.login) { store in
-        NavigationView {
-          LoginView(store: store)
+    SwitchStore(self.store) { state in
+      switch state {
+      case .login:
+        CaseLet(state: /TicTacToe.State.login, action: TicTacToe.Action.login) { store in
```

Comment:
Looking at this diff I'm wondering what function the `SwitchStore` provides

* Instead of `SwitchStore`+`CaseLet` could an `IfLetStore` perform the same work? 
* If you want completeness over all cases, use a ViewStore scoped to state?

Then, thinking about this in terms of navigation, we're often peeling off a single enum case in different parts of the view, not necessarily constructing a complete switch statement in one place. Should we think about other conditional views the same way? Again, maybe `IfLetStore` is the only tool needed?

Reply:
> If you want completeness over all cases, use a ViewStore scoped to state?

If done naively, such a view store would observe all mutations to child state in each associated value, which could easily lead to performance problems and other glitches. `SwitchStore` automatically de-dupes to only re-render when the case changes using the Swift runtime metadata.